src/handle-github/branch.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Suggester} from '../application';
+const mainBranch = 'master';
+
+function getBranchFromRef(ref: string) {
+  return ref.split('/')[2];
+}
+
+function getUniqueBranchName(
+  branchName: string,
+  conflictingNames: string[]
+): string {
+  if (!conflictingNames.length) return branchName;
+  else if (conflictingNames.length === 1) return `${branchName}-2`;
+  conflictingNames.sort();
+  const lastName = conflictingNames[conflictingNames.length - 1];
+  const splitName = lastName.split('-');
+  const highestCount = Number(splitName[splitName.length - 1]) + 1;
+  return `${branchName}-${highestCount}`;
+}
+
+/**
+ * Create a GitHub branch given a destination repo and destination owner/organization
+ * @param app the Suggester instance
+ * @returns the base SHA for subsequent commits to be based off, and the branch name to apply commits to
+ */
+async function branch(app: Suggester) {
```

Comment:
Let's name this `createBranch` as `branch` can also be a noun. Perhaps add an optional argument for specifying the base branch name which has a default value.

Reply:
Since Suggester is a class, does this belong as a method on Suggester instead?

Reply:
Is it a concern that Suggester would then be a very massive class?

Reply:
Removed all classes